### PR TITLE
[GPU] fp16-int8 mixed precision

### PR DESCRIPTION
**Pull request by isanghao**
### Details:
 - fp16-int8 mixed precision is implemented
- Allow quantization fusion into previous conv if the it is onednn conv
- Implemented cross-precision and cross-layout configuration for onednn conv
- Select layout based on output of primitives, not input
- Do not enforce fp32 for mixed-precision network

### Tickets:
 - 65603

### CODE REVIEW

File: src/plugins/intel_gpu/src/graph/layout_optimizer.cpp [hunk 1]
```diff
@@ -913,10 +921,10 @@ layout layout_optimizer::get_expected_layout(layout const& current_layout,
         /* ***************************** OneDNN impls format selection part ****************************** */
         bool valid_grouped = !is_dw && prim->groups > 1 && (ofm_per_group % compute_block == 0 && ifm_per_group % compute_block == 0);
         // TODO: uncomment this code when corresponding fsv32 optimizations inside clDNN will be implemented
-        // bool i8_u8_output = output_layout.data_type == data_types::u8 || output_layout.data_type == data_types::i8;
+        bool i8_u8_output = output_layout.data_type == data_types::u8 || output_layout.data_type == data_types::i8;
```
Comment (lznamens) [inline src/plugins/intel_gpu/src/graph/layout_optimizer.cpp:924]: I suppose the comment above can be removed.

**Review by lznamens — APPROVED** [review]
LGTM if you verified that there are no perf regressions.

**Comment by isanghao** [comment]
> sorry, didn't see do not review

@lznamens  Thanks for the reply. It is OK to review. It is tagged as do_not_review because I am seeing some perf drop.

File: src/plugins/intel_gpu/src/graph/layout_optimizer.cpp (hunk 1 above)
Comment (isanghao) [reply src/plugins/intel_gpu/src/graph/layout_optimizer.cpp:947]: Applied and now this PR is ready for review. Please have a look!

File: src/plugins/intel_gpu/src/graph/layout_optimizer.cpp [hunk 2]
```diff
@@ -1255,6 +1286,14 @@ impl_types layout_optimizer::get_preferred_impl_type(program_node& node, format
         if (input_layout.format.dimension() != output_layout.format.dimension()) {
             preferred_impl = impl_types::ocl;
         }
+
+        // For mixed precision case, onednn is slower than cldnn
+        if (input_layout.format == format::b_fs_yx_fsv16 && (input_layout.data_type == data_types::u8 || input_layout.data_type == data_types::i8))
```
Comment (lznamens) [inline src/plugins/intel_gpu/src/graph/layout_optimizer.cpp:1291]: Please use local variables with short names like `input_fmt` (defined in a code line 1265), `input_dt` in this new block of the code and in the code above which is related to `node.is_type<reorder>` and which you didn't change.

Comment (isanghao) [reply src/plugins/intel_gpu/src/graph/layout_optimizer.cpp]: Applied, thanks. I defined `data_type_traits::is_i8_u8` as a helper function, too.

**Pull request merged**